### PR TITLE
Enable style_src: unsafe_inline csp to about page

### PR DESCRIPTION
**Pull request by tribela**
So that admin can embed css on `/about` and `/about/more` page

### CODE REVIEW

File: config/initializers/content_security_policy.rb [hunk 1]
```diff
@@ -51,3 +51,7 @@ def host_to_url(str)
   p.script_src :self, :unsafe_inline, assets_host
   p.style_src  :self, :unsafe_inline, assets_host
 end
+
+AboutController.content_security_policy do |p|
+  p.style_src :self, :unsafe_inline, assets_host
+end
```
Comment (ThibG) [inline config/initializers/content_security_policy.rb:57]: I am not sure I agree with those changes, but I think the build failures would be avoided by putting this directive directly in `app/controllers/about_controller.rb`

**Review by ClearlyClaire — DISMISSED** [review]

File: config/initializers/content_security_policy.rb (hunk 1 above)
Comment (Kjwon15) [reply config/initializers/content_security_policy.rb:57]: It is failed by `JsonldHelper`
https://circleci.com/gh/tootsuite/mastodon/134916?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

Comment (ThibG) [reply config/initializers/content_security_policy.rb:57]: Yeah, I think that failure will be avoided if you move that rule out of this file and in `app/controllers/about_controller.rb`.

My understanding is that `JsonLdHelper` works as is thanks to the rules added in `config/initializers/inflections.rb` but that gets loaded *after* `config/initializers/content_security_policy.rb`.

Comment (Kjwon15) [reply config/initializers/content_security_policy.rb:57]: But i think it's better when gather all CSP into one file and fix #13874

Comment (ThibG) [reply config/initializers/content_security_policy.rb:57]: having all CSP rules in one place kind of makes sense, but also having them in more specific places does have sense too (`app/controllers/statuses_controller.rb` already does that for the embed endpoint)

**Comment by Kjwon15** [comment]
Uh... assets_hosts..

File: config/initializers/content_security_policy.rb (hunk 1 above)
Comment (Kjwon15) [reply config/initializers/content_security_policy.rb]: It seems not inheritable from original CSP config.. and `assets_host` is the problem.

**Comment by ThibG** [comment]
Don't want to mark as approved because I'm not completely sure about the idea of adding 'unsafe-inline' on those pages, but otherwise the PR looks fine, thank you!

**Comment by Gargron** [comment]
Is custom CSS not enough to customize the about pages without relying on inline styles?

**Comment by SuperSandro2000** [comment]
@Kjwon15 bump

**Pull request closed without merging**